src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -345,6 +365,9 @@ public void init(AdaptorContext context) throws Exception {
 
     uniqueKey = ukBuilder.build();
     log.config("primary key: " + uniqueKey);
+
+    docIdLastModifiedMap =
+        CacheBuilder.newBuilder().maximumSize(1000000).build();
```

Comment:
We might want to set an initial capacity. The (undocumented) default is 16 (though I think it might be 40 in practice), which is pretty small, and will lead to lots of allocations to grow the cache (done by doubling). Maybe initialCapacity(10000)?